src/sentry/templates/sentry/integrations/jira-config.html

```diff
@@ -23,38 +23,79 @@
-  <div class="aui-message aui-message-success">
-    <p class="title">
-      <strong>Saved!</strong>
-    </p>
-    <p>
-      The Sentry Jira integration is now enabled for the selected
-      organizations. Return to your sentry organization to configure the Jira
-      integration for your organization.
-    </p>
-  </div>
-  {% endif %}
+  {% if login_required %}
+    <div class="aui-message aui-message-info">
+        <p>Please login to your Sentry account to access the Sentry Add-on configuration</p>
+    </div>
+    <div class="signin">
+      <a class="aui-button aui-button-default" href="{{ login_url }}" target="_blank">
+        Sign In to Sentry
+      </a>
+    </div>
+  {% else %}
+    {% if completed %}
+    <div class="aui-message aui-message-success">
+      <p class="title">
+        <strong>Saved!</strong>
+      </p>
+      <p>
+        The Sentry Jira integration is now enabled for the selected
+        organizations. Return to your sentry organization to configure the Jira
+        integration for your organization.
+      </p>
+    </div>
+    {% endif %}
 
-  <form class="aui top-label" action="" method="post">
-    {% csrf_token %}
-    {% for field in form %}
-      <div class="field-group top-label">
-        {{ field.errors }}
-        {{ field.label_tag }} {{ field }}
-        <div class="description">{{ field.help_text }}</div>
+    <form class="aui top-label" action="" method="post">
+      {% csrf_token %}
+      {% for field in form %}
+        <div class="field-group top-label">
+          {{ field.errors }}
+          {{ field.label_tag }} {{ field }}
+          <div class="description">{{ field.help_text }}</div>
+        </div>
+      {% endfor %}
+      <div class="field-group">
+        <button class="aui-button aui-button-primary" type="submit">
+            Save Settings
+        </button>
       </div>
-    {% endfor %}
-    <div class="field-group">
-      <button class="aui-button aui-button-primary" type="submit">
-          Save Settings
-      </button>
-    </div>
-  </form>
+    </form>
+  {% endif %}
 </body>
+{% if login_required %}
+  <script type="text/javascript">
+    (function() {
+      var hidden, visibilityChange;
+      if (typeof document.hidden !== 'undefined') { // Opera 12.10 and Firefox 18 and later support
+        hidden = 'hidden';
+        visibilityChange = 'visibilitychange';
+      } else if (typeof document.mozHidden !== 'undefined') {
+        hidden = 'mozHidden';
+        visibilityChange = 'mozvisibilitychange';
+      } else if (typeof document.msHidden !== 'undefined') {
+        hidden = 'msHidden';
+        visibilityChange = 'msvisibilitychange';
+      } else if (typeof document.webkitHidden !== 'undefined') {
+        hidden = 'webkitHidden';
+        visibilityChange = 'webkitvisibilitychange';
+      }
+      document.addEventListener(visibilityChange, function() {
+        // see if they've logged in while away from tab
+        if (!document[hidden]) {
```

Comment:
https://caniuse.com/#feat=pagevisibility I would drop all this boilerplate. I didn't use any of it here https://github.com/getsentry/sentry/blob/962badb2b6b2b132b032d8254fc0f0568c0e505c/src/sentry/static/sentry/app/components/asyncComponent.jsx#L51